src/pytest_bdd/gherkin_parser.py

```diff
@@ -8,7 +12,68 @@
 from gherkin.token_scanner import TokenScanner
 
 from . import exceptions
-from .types import STEP_TYPES
+
+if typing.TYPE_CHECKING:
+    from typing import Self
+
+
+ERROR_PATTERNS = [
```

Comment:
absolutely appreciate the effort you put to represent the original exception! I see though that some of them are not tested, would you mind add simple tests for them?
Of course, tests didn't exist before either, so if you don't feel like it it's perfectly fine

Reply:
Added tests.  Some I realise don't happen as I expected in the parser, so updated exceptions list and ensured at least one test for each.